packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -39,15 +39,29 @@ export interface FluidAppInsightsLoggerConfig {
 
 /**
  * Object used with an {@link FluidAppInsightsLoggerConfig}
- * to define logic for filtering of telemetry events
+ * to define a filter with logic for matching it to telemetry events.
+ * Filters can include either a category, namespace or both types of filter; A valid filter must have at least one defined.
  *
  * @public
  */
 export interface TelemetryFilter {
 	/**
 	 * The category of telemetry event that this filter applies to
 	 */
-	category: TelemetryEventCategory;
+	category?: TelemetryEventCategory;
+	/**
+	 * The namespace pattern to filter telemetry events.
+	 * For example, "perf:latency:*" will match any namespace starting with "perf:latency:"
+	 * This is a regex pattern.
+	 */
+	namespacePattern?: string;
+	/**
+	 * A list of namespace patterns to explicitly exclude from the filter.
+	 * For example, if you have a namespacePattern of "perf:latency:*" but want to exclude
+	 * events from "perf:latency:ops", you would add "perf.latency.ops" to this list.
+	 * These are regex patterns.
+	 */
+	namespacePatternExceptions?: string[];
```

Comment:
I'm not sure this needs to be separate from `namespacePattern`. Whether it's a glob pattern or regex, users should be able to specify exceptions inline.

Reply:
Since I swapped from Regex to .startsWith per alejandros suggestions, we'll have to keep this since we'll need an explicit place for exceptions.